test/io/fixtures.sql

```diff
@@ -213,3 +213,10 @@ create or replace function multiple_func_settings_test() returns setof record as
 $$ language sql
 set work_mem = '5000'
 set statement_timeout = '10s';
+
+create or replace function not_hoisted_should_not_apply_test() returns setof record as $$
+  select current_setting('work_mem',false) as work_mem,
+         current_setting('statement_timeout',false) as statement_timeout;
+$$ language sql
+set work_mem = '5000'
+set statement_timeout = '5s';
```

Comment:
@wolfgangwalther How about this?
```sql
-- returns computed fields
create or replace function not_hoisted_should_not_apply()
returns table (
  work_mem text,
  statement_timeout text
)
as $$
begin
  return query
          select
            current_setting('work_mem',false) as work_mem,
            current_setting('statement_timeout',false) as statement_timeout;
end;
$$ language plpgsql
set work_mem = '5000'
set statement_timeout = '5s';

-- calls the above function
create or replace function test_not_hoisted_should_not_apply()
returns text
as $$
  select work_mem || ' ' || statement_timeout
  from not_hoisted_should_not_apply();
$$ language sql
```
Calling `test_not_hoisted_should_not_apply` yields `5000kb 5s` which is strange because only `work_mem` is hoisted but `statement_timeout` is still being applied. Is this right way to write computed fields?

Reply:
@wolfgangwalther Thanks a lot! It worked.